specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json

```diff
@@ -1754,6 +1760,12 @@
             "schema": {
               "$ref": "#/definitions/LocalUser"
             }
+          },
+          "default": {
```

Comment:
From .net SDK/Pwoershell point of view, We should remove the "default" responds handling part, since:
1. Without  the default part, the error will be handled by default CloudException in .net SDK, and the error details will be in exception message. 
2. With the default part, the error will be handle by storage defined "#/definitions/ErrorResponse", the returned exception in .net SDK will only has error code in exception message, but won't has detail error message in the exception message. It will be difficult for customer to find the detail error message, as they need go into the exception properties and  try to find which property contains the error.

As the default error handling can handle error with following body very well. I would suggest:

-  if the error body of an API have following format, we should not add "default" responds part for it.
- only add default responds handling for error , when the error body is not in following format.

```
Body:
{
  "error": {
    "code": "***",
    "message": "****"
  }
}
```

For you reference, following are errors from .net SDK:
Error from API without default responds section:
```
Microsoft.Rest.Azure.CloudException: An operation is currently performing on this storage account that requires exclusive access.
   at Microsoft.Azure.Management.Storage.StorageAccountsOperations.<DeleteWithHttpMessagesAsync>d__7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.Storage.StorageAccountsOperationsExtensions.<DeleteAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.Storage.StorageAccountsOperationsExtensions.Delete(IStorageAccountsOperations operations, String resourceGroupName, String accountName)
   at Microsoft.Azure.Commands.Management.Storage.RemoveAzureStorageAccountCommand.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
```
Error from API with default responds section:
```
Microsoft.Azure.Management.Storage.Models.ErrorResponseException: Operation returned an invalid status code 'BadRequest'
   at Microsoft.Azure.Management.Storage.StorageAccountsOperations.<BeginHierarchicalNamespaceMigrationWithHttpMessagesAsync>d__23.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Azure.Management.Storage.StorageAccountsOperations.<HierarchicalNamespaceMigrationWithHttpMessagesAsync>d__17.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Azure.Management.Storage.StorageAccountsOperationsExtensions.<HierarchicalNamespaceMigrationAsync>d__25.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.Azure.Management.Storage.StorageAccountsOperationsExtensions.HierarchicalNamespaceMigration(IStorageAccountsOperations operations, String resourceGroupName, String accountName, String requestType)
   at Microsoft.Azure.Commands.Management.Storage.InvokeAzureStorageAccountHierarchicalNamespaceUpgradeCommand.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
```

Reply:
Hi Wei, I added it back because it failed the validation if no default response is defined. 

"paths":[{"tag":"Location","path":"https://github.com/Azure/azure-rest-api-specs/blob/0c3632266c799502dccecd249377c606b3be9ab3/specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/storage.json#L680"}]},{"type":"Result","level":"Error","message":"The response is defined but without a default error response implementation.Consider adding it.'","code":"RequiredDefaultResponse","id":"R4010","docUrl":"https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#R4010"

Reply:
@jianyexi 
Why we have this check?
The check seems not fair , espicially for .net SDK. It will hide the real failure reason in deep level of excpetion, instead of show it in the exception message. 
Could we remove the check ?

Reply:
This is required by ARM , may be dotnet SDK is special , for a consistent user experience for all the SDKs, we should respect default response

Reply:
@jianyexi
Could you please give the reason why this is required by ARM?
currently from .net SDK point of view, the part is not necessary, and really take issue to .net SDK.